main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2289,13 +2290,20 @@ object Classpaths {
       inTask(updateSbtClassifiers)(
         Seq(
           externalResolvers := {
+            val boot = bootResolvers.value
             val explicit = buildStructure.value
               .units(thisProjectRef.value.build)
               .unit
               .plugins
               .pluginData
               .resolvers
-            explicit orElse bootRepositories(appConfiguration.value) getOrElse externalResolvers.value
+            // https://github.com/sbt/sbt/issues/4408
+            (explicit, boot) match {
+              case (Some(x), Some(boot)) => (x.toVector ++ boot.toVector).distinct
```

Comment:
This really doesn't matter, but this would be _slightly_ more clear if `x` was renamed `explicit` since you're already shadowing `boot` anyway.